### PR TITLE
Chore: Update Jitsi to 3.6.0

### PR DESCRIPTION
**Pull request by diegolmello**
## Proposed changes

Updates Jitsi SDK to 3.6.0.
All forks were updated.

## Issue(s)	

Depends on:
https://github.com/RocketChat/react-native-jitsi-meet/pull/9
https://github.com/RocketChat/jitsi-maven-repository/pull/2
https://github.com/RocketChat/jitsi-meet-ios-sdk-releases/pull/5
https://github.com/RocketChat/jitsi-meet/pull/14

### Starting a call
- Go to any room actions
- Tap Voice call/Video call (we should test both)

### Joining from message
- Start a call from web
- A "Call started by" message should be sent to the room
- Join by tapping "Click to join"
- _Known issue:_ During development, I found that if you start this process from mobile and try to join on the web, it won't work. We need to test that and check if there's an open issue on the web repo.

### Joining from push notification
- Start a call from web
- A push notification should be sent to your device
- Join by tapping the push notification

### Test Jitsi features
- We need to test Jitsi features making sure everything is working properly (mute, turn off cam, etc)
- Some of them might be flaky or not work because of Jitsi itself, but it's good to be aware of
- 
### Dark mode
- Enable dark mode on your device
- Everything should be visible

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] Improvement (non-breaking change which improves a current function)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update (if none of the other choices apply)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat) doc
- [x] I have signed the [CLA](https://cla-assistant.io/RocketChat/Rocket.Chat.ReactNative)
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [ ] I have added necessary documentation (if applicable)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: android/build.gradle [hunk 1]
```diff
@@ -17,8 +17,8 @@ buildscript {
         kotlin_version = "1.3.50"
         supportLibVersion = "28.0.0"
         libre_build = !(isPlay.toBoolean())
-        jitsi_url = isPlay ? "https://github.com/RocketChat/jitsi-maven-repository/raw/master/releases" : "https://github.com/RocketChat/jitsi-maven-repository/raw/libre/releases"
-        jitsi_version = isPlay ? "2.10.2" : "2.10.0-libre"
+        jitsi_url = isPlay ? "http://github.com/RocketChat/jitsi-maven-repository/raw/3.7.0/releases" : "https://github.com/jitsi/jitsi-maven-repository/raw/libre/releases"
```
Comment (diegolmello) [inline android/build.gradle:20]: We're not actively maintaining libre, so I just changed the URL to match, but there's no point to test at this time.

File: app/views/JitsiMeetView.js [hunk 2]
```diff
@@ -30,39 +31,45 @@ class JitsiMeetView extends React.Component {
...
 			}
-		}, 1000);
+		};
+	}
+
+	componentDidMount() {
+		const { route } = this.props;
+		const { userInfo } = this.state;
+
+		if (isIOS) {
```
Comment (diegolmello) [inline app/views/JitsiMeetView.js:55]: I'm calling Jitsi on iOS the old way and Android using props.
For some reason I didn't understand, Android is not working this way anymore.

File: app/views/JitsiMeetView.js [hunk 3]
```diff
@@ -83,18 +92,20 @@ class JitsiMeetView extends React.Component {
...
+		const { route } = this.props;
+		const onlyAudio = route.params?.onlyAudio ?? false;
+		const options = isAndroid ? { url: this.url, userInfo, audioOnly: onlyAudio } : null;
 		return (
 			<RNJitsiMeetView
 				onConferenceTerminated={this.onConferenceTerminated}
 				onConferenceJoined={this.onConferenceJoined}
-				style={sharedStyles.container}
+				style={StyleSheet.absoluteFill}
+				options={options}
```
Comment (diegolmello) [inline app/views/JitsiMeetView.js:108]: This is what triggers the call on Android from now on.

File: ios/Podfile [hunk 4]
```diff
@@ -15,7 +15,7 @@ end
 
 abstract_target 'defaults' do
   # force use our own JitsiMeetSDK
-  pod 'JitsiMeetSDK', :git => 'https://github.com/RocketChat/jitsi-meet-ios-sdk-releases.git'
+  pod 'JitsiMeetSDK', :git => 'https://github.com/RocketChat/jitsi-meet-ios-sdk-releases.git', :branch => '3.7'
```
Comment (diegolmello) [inline ios/Podfile:18]: We need to remove this branch and reinstall pods before merging

**Review by gerzonc — APPROVED** [review]
LGTM :)

**Comment by ankar84** [comment]
> * _Known issue:_ During development, I found that if you start this process from mobile and try to join on the web, it won't work. We need to test that and check if there's an open issue on the web repo.

Yeap, it's a front-end bug, that will be fixed in next release, I guess

https://github.com/RocketChat/Rocket.Chat/issues/22845

**Comment by ankar84** [comment]
> ### Known issue
> * Sometimes Jitsi's "End call" button doesn't work (it just grey out)
> * When that happens, I needed to tap several times or wait until the call was ended
> * I was unable to find how to reproduce that 100%, but we need to test that and evaluate if it's a critical issue (if it happens too often, if the call is ended properly, etc)

And that's really strange. I didn't get that kind if bug using Jitsi meet application which on same 3.7.0 SDK version.

But I will wait RCRN Experimental beta to test it with our production Jitsi deployment and left a feedback for sure!

File: .circleci/config.yml [hunk 5]
```diff
@@ -64,13 +64,13 @@ update-fastlane-android: &update-fastlane-android
 
 save-gradle-cache: &save-gradle-cache
   name: Save gradle cache
-  key: android-{{ checksum "android/build.gradle" }}-{{ checksum  "android/app/build.gradle" }}
+  key: android-v2-{{ checksum "android/build.gradle" }}-{{ checksum  "android/app/build.gradle" }}
```
Comment (diegolmello) [inline .circleci/config.yml:67]: Remove this after changing gradle to correct url

**Pull request merged**